src/features/chain-connection/WalletConnection.tsx

```diff
@@ -1,32 +1,91 @@
-import { useAccount, useConnect, useEnsName } from "wagmi";
+import { useAccount, useEnsName, useDisconnect, useConnect, Connector } from "wagmi";
 import { Button } from "../common/Button";
-import { InjectedConnector } from "wagmi/connectors/injected";
 import { twJoin } from "tailwind-merge";
 import truncateEthAddress from "truncate-eth-address";
+import { VscChromeClose } from "react-icons/vsc";
+import { useEffect } from "react";
 
-// TODO: this component should actually open a modal instead of defaulting to
-// the injected connector
-
-export const WalletConnection = ({}: {}) => {
-  const { address, isConnected } = useAccount();
+const WalletConnection = () => {
+  const { address, isConnected, isConnecting } = useAccount();
   const { data: ensName } = useEnsName({ address });
+  const { connect, connectors, isLoading, pendingConnector } = useConnect()
+  const { disconnect } = useDisconnect()
+
+  const dialog = document.querySelector('dialog')
```

Comment:
This could select a different `dialog` element, if someone implemented a modal elsewhere.

To get a reference to a DOM element in your component you should [use `useRef`](https://react.dev/reference/react/useRef#manipulating-the-dom-with-a-ref)